### PR TITLE
MDEV-21869: Various spelling fixes

### PR DESCRIPTION
**Pull request by ottok**

### CODE REVIEW

**Comment by an3l** [comment]
@ottok can you please squash in 1 commit this?

**Comment by ottok** [comment]
For the record, builbot failures seem sporadic/unrelated:
image

**Comment by ottok** [comment]
Squashed into 1 commit as requested.

**Comment by an3l** [comment]
Thanks Otto, do you want to add `MDEV-21869:` to you commit message? :)

**Comment by ottok** [comment]
Will "MDEV-21869:" in title automatically close that issue? Because this commit is not enough to close that issue. Shall I put in body "Related: MDEV-21869"?

**Comment by an3l** [comment]
Setting the name in commit message, shouldn't close automatically the issue in jira (after push), but it is a valid pointer that jira issue should be closed. 
AFAIK about automatically closing the tickets,it is for PR's(or issues), by specifying `Closes #` with number of PR in commit body (although from time to time this doesn't work well).
As you said that the current commit is not enough, I think it is better what you suggested with comment in body: `Related to MDEV-21769`.

**Comment by ottok** [comment]
Commit message amended and re-pushed.

File: sql/sql_select.cc [hunk 1]
```diff
@@ -14928,16 +14928,16 @@ bool Item_func_eq::check_equality(THD *thd, COND_EQUAL *cond_equal,
     The function performs the substitution in a recursive descent by
     the condtion tree, passing to the next AND level a chain of multiple
     equality predicates which have been built at the upper levels.
-    The Item_equal items built at the level are attached to other 
+    The Item_equal items built at the level are attached to other
     non-equality conjucts as a sublist. The pointer to the inherited
     multiple equalities is saved in the and condition object (Item_cond_and).
-    This chain allows us for any field reference occurence easyly to find a 
-    multiple equality that must be held for this occurence.
+    This chain allows us for any field reference occurrence easyly to find a
```
Comment (cvicentiu) [inline sql/sql_select.cc]: occurrence easily

File: sql/opt_range.cc [hunk 2]
```diff
@@ -5757,7 +5757,7 @@ bool prepare_search_best_index_intersect(PARAM *param,
     {
       idx_scan.add("chosen", true);
       if (!*scan_ptr)
-        idx_scan.add("cause", "first occurence of index prefix");
+        idx_scan.add("cause", "first occurrence of index prefix");
```
Comment (cvicentiu) [inline sql/opt_range.cc]: Now this is an interesting find!

No test cases for this code. We should have seen this change reflect as a test case failure.

Ok to push however. Writing a (stable) test case for this is not trivial.

File: sql/sql_select.cc (hunk 1 above)
Comment (cvicentiu) [reply sql/sql_select.cc]: @an3l please adjust the commit to also fix the "easily" word, then ok to push.

Comment (an3l) [reply sql/sql_select.cc]: @ottok do you want to fix this please, because of github statistics (merged over closed)?

Comment (ottok) [reply sql/sql_select.cc]: Done.

**Comment by ottok** [comment]
Updated as requested and pushed. Once the latest round of test build pass, I'll merge this on 10.5.

**Comment by ottok** [comment]
Builds can't pass since Travis-CI has been broken with missing dependencies and failing s390x builds. Will fix those in another set of commits..

**Pull request closed without merging**